### PR TITLE
add tests and docs for using custom service account in dataflow flex template job

### PR DESCRIPTION
**Pull request by xingao267**
fixes https://github.com/hashicorp/terraform-provider-google/issues/7658

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dataflow: added documentation about using `parameters` for custom service account and other pipeline options to `google_dataflow_flex_template_job`
```

### CODE REVIEW

**Comment by xingao267** [comment]
I tested it locally. It seems the dataflow jobs is created successfully with the custom service account, but test failed with the following error:

```
$ make testacc TEST=./google-beta TESTARGS='-run=TestAccDataflowFlexTemplateJob_withServiceAccount'
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccDataflowFlexTemplateJob_withServiceAccount -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccDataflowFlexTemplateJob_withServiceAccount
=== PAUSE TestAccDataflowFlexTemplateJob_withServiceAccount
=== CONT  TestAccDataflowFlexTemplateJob_withServiceAccount
    provider_test.go:266: Step 1/1 error: Check failed: Check 2/2 error: Error getting dataflow job instance template: no instance template found for dataflow job "2020-11-24_09_30_40-2187942558452290776"
--- FAIL: TestAccDataflowFlexTemplateJob_withServiceAccount (783.03s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-google-beta/google-beta	783.237s
FAIL
make: *** [GNUmakefile:15: testacc] Error 1
```
@rileykarson could you help take a look at this PR?

**Review by rileykarson — COMMENTED** [review]
I think the error comes from `testAccDataflowJobGetGeneratedInstanceTemplate`. I'd suggest running the test with `TF_LOG=DEBUG` set, and looking at the HTTP requests/responses.

Also, can you add the field to the documentation? https://github.com/GoogleCloudPlatform/magic-modules/blob/master/third_party/terraform/website/docs/r/dataflow_flex_template_job.html.markdown

**Comment by xingao267** [comment]
> I think the error comes from `testAccDataflowJobGetGeneratedInstanceTemplate`. I'd suggest running the test with `TF_LOG=DEBUG` set, and looking at the HTTP requests/responses.
> 
> Also, can you add the field to the documentation? https://github.com/GoogleCloudPlatform/magic-modules/blob/master/third_party/terraform/website/docs/r/dataflow_flex_template_job.html.markdown

Thanks! Fixed the test. For flex template job, I need to get the SA info directly from the VM but not VM template.

File: third_party/terraform/resources/resource_dataflow_flex_template_job.go.erb [hunk 1]
```diff
@@ -41,6 +41,14 @@ func resourceDataflowFlexTemplateJob() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"zone": {
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (rileykarson) [inline third_party/terraform/resources/resource_dataflow_flex_template_job.go.erb:46]: We probably want `Computed: true` as well, or else we'll see a diff when the provider zone is used.

File: third_party/terraform/resources/resource_dataflow_flex_template_job.go.erb [hunk 2]
```diff
@@ -115,8 +123,11 @@ func resourceDataflowFlexTemplateJobCreate(d *schema.ResourceData, meta interfac
 		return err
 	}
 
+	zone, _ := getZone(d, config)
```
Comment (rileykarson) [inline third_party/terraform/resources/resource_dataflow_flex_template_job.go.erb:126]: This field is optional in the API, and that's why we can ignore the error right? Mind commenting as much?

Comment (xingao267) [reply third_party/terraform/resources/resource_dataflow_flex_template_job.go.erb:129]: Done.

**Review by rileykarson — COMMENTED** [review]
Looking at the "all commits" view, it appears that the actual resource changes got reverted at some point?

**Comment by xingao267** [comment]
@rileykarson according to https://github.com/hashicorp/terraform-provider-google/issues/7658#issuecomment-736122829, it turned out we can specify custom SA in the parameters block. So I reverted the actual resource changes and made this PR just adding tests and docs for how to use custom SA.

**Review by rileykarson — APPROVED** [review]
LGTM- just running through the test pipeline before merging.

/gcbrun

**Comment by rileykarson** [comment]
Well, that didn't appear to have taken. I'll try again!

/gcbrun

**Comment by xingao267** [comment]
@rileykarson I can't see the builds, do you know why it is failing?

**Comment by rileykarson** [comment]
Just a timeout! The VCR test run I triggered should supersede it. It doesn't look like the second phase kicked off- I'll do that manually on Monday.

**Comment by xingao267** [comment]
@rileykarson any updates on this?

**Comment by rileykarson** [comment]
Sorry! Heavy load of interrupts on my part this last week kept me distracted. Started a run now.

For my own benefit, that's https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaMmUpstream/162586?buildTab=overview

**Review by rileykarson — APPROVED** [review]

**Comment by xingao267** [comment]
@rileykarson it looks like the documentation change included in this PR did not get released to https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dataflow_flex_template_job. Did I miss anything?

**Review by rileykarson — COMMENTED** [review]
That's expected! We had a fairly substantial gap between releases (https://github.com/hashicorp/terraform-provider-google/issues/7998)

It should be available today.

**Comment by xingao267** [comment]
@rileykarson got it, thanks for explaining!

**Pull request merged**